src/helpers/mathjax.coffee

```diff
@@ -34,10 +34,15 @@ typesetMath = (root) ->
       node.textContent = "#{MATH_MARKER_BLOCK}#{formula}#{MATH_MARKER_BLOCK}"
     else
       node.textContent = "#{MATH_MARKER_INLINE}#{formula}#{MATH_MARKER_INLINE}"
-    window.MathJax.Hub.Queue(['Typeset', MathJax.Hub, node])
     # mark node as processed
     node.classList.add('math-rendered')
 
+  # convert NodeList to Array
+  node_array = Array.prototype.slice.call(nodes)
+
+  # submit all nodes at once for optimal rendering performance
+  window.MathJax.Hub.Queue(['Typeset', MathJax.Hub, node_array])
+
```

Comment:
Wow if this works, it should indeed be more efficient.  I wasn't aware that you could pass an array into `Hub.Queue`

You could just use underscore's _.toArray though:
`window.MathJax.Hub.Queue(['Typeset', MathJax.Hub, _.toArray(nodes)])`

That's assuming that `Hub.Queue` doesn't work with just a  `NodeList` ?